src/modules/logicModule/LM_PC_FundingPot_v1.sol

```diff
@@ -0,0 +1,1438 @@
+ *
+ *          - Access Control.
+ *            Multiple access criteria types:
+ *            - Allowlist-based access.
+ *            - NFT ownership verification.
+ *            - Merkle proof validation.
+ *            - Open access.
+ *
+ *          - Contribution Management.
+ *            - Personal contribution caps.
+ *            - Round-level caps.
+ *            - Global accumulative caps across rounds.
+ *            - Configurable contribution time windows.
+ *
+ *          - Automated Processing.
+ *            - Automatic round closure based on time or cap.
+ *            - Post-round hook execution.
+ *            - Payment order creation for contributors.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @custom:version  v1.0.0.
+ *
+ * @custom:inverter-standard-version v0.1.0.
+ *
+ * @author  33Audits.
+ */
+contract LM_PC_FundingPot_v1 is
+    ILM_PC_FundingPot_v1,
+    ERC20PaymentClientBase_v2
+{
+    // -------------------------------------------------------------------------
+    // Libraries
+
+    using SafeERC20 for IERC20;
+
+    // -------------------------------------------------------------------------
+    // ERC165
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        virtual
+        override(ERC20PaymentClientBase_v2)
+        returns (bool)
+    {
+        return interfaceId_ == type(ILM_PC_FundingPot_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    // --------------------------------------------------------------------------
+    // Constants
+
+    /// @notice The role that allows creating funding rounds.
+    bytes32 public constant FUNDING_POT_ADMIN_ROLE = "FUNDING_POT_ADMIN";
+
+    /// @notice The payment processor flag for the start timestamp.
+    uint8 internal constant FLAG_START = 1;
+
+    /// @notice The payment processor flag for the cliff timestamp.
+    uint8 internal constant FLAG_CLIFF = 2;
+
+    /// @notice The payment processor flag for the end timestamp.
+    uint8 internal constant FLAG_END = 3;
+
+    /// @notice The maximum valid access criteria ID.
+    uint8 internal constant MAX_ACCESS_CRITERIA_TYPE = 4;
+
+    // -------------------------------------------------------------------------
+    // State
+
+    /// @notice The current round count.
+    uint32 private roundCount;
+
+    /// @notice Stores all funding rounds by their unique ID.
+    mapping(uint32 => Round) private rounds;
+
+    /// @notice Stores all access criteria privilages by their unique ID.
+    mapping(
+        uint32 roundId
+            => mapping(uint8 accessCriteriaId_ => AccessCriteriaPrivileges)
+    ) private roundIdToAccessCriteriaIdToPrivileges;
```

Comment:
why put this in a separate mapping? is it not cleaner to have the priviliges part of the accessCriteria object instead of createing a separate mapping?